sdk/cosmos/azure-cosmos/azure/cosmos/_base.py

```diff
@@ -227,7 +227,10 @@ def GetHeaders(  # pylint: disable=too-many-statements,too-many-branches
         # else serialize using json dumps method which apart from regular values will serialize None into null
         else:
             # single partitioning uses a string and needs to be turned into a list
-            if isinstance(options["partitionKey"], list) and options["partitionKey"]:
+            is_tuple_or_list = True in (isinstance(options["partitionKey"], list),
+                                            isinstance(options["partitionKey"], tuple))
+
+            if is_tuple_or_list and options["partitionKey"]:
```

Comment:
`json.dumps` works with almost identical regardless of `separators` for list, tuple, the only difference is that if we add the `separators`, it does not add spaces after comma. Is this really necessary? 

Also, we could simplify the conditions to be `isinstance(options["partitionKey"], (list, tuple))`. We don't need extra variable.